eth/catalyst/api.go

```diff
@@ -102,47 +105,57 @@ func (api *ConsensusAPI) ForkchoiceUpdatedV1(update beacon.ForkchoiceStateV1, pa
 		// that should be fixed, not papered over.
 		header := api.remoteBlocks.get(update.HeadBlockHash)
 		if header == nil {
-			log.Warn("Forkcoice requested unknown head", "hash", update.HeadBlockHash)
-			return beacon.ForkChoiceResponse{Status: beacon.INVALID}, errors.New("head hash never advertised")
+			log.Warn("Forkchoice requested unknown head", "hash", update.HeadBlockHash)
+			return beacon.STATUS_SYNCING, nil
 		}
 		// Header advertised via a past newPayload request. Start syncing to it.
 		// Before we do however, make sure any legacy sync in switched off so we
 		// don't accidentally have 2 cycles running.
 		if merger := api.eth.Merger(); !merger.TDDReached() {
 			merger.ReachTTD()
 			api.eth.Downloader().Cancel()
+			// Shutdown gpu miner
```

Comment:
This feels like a time bomb. You could have called `ending.Close()`, since that just delegates to the legacy engine to close, yet your type checks reveals that you don't want a generic *close* here, since you don't really want to stop the engine, just stop ethash pow mining.

I think it's dangerous in itself to Close here because it may or may not release required resources (e.g. ethash verification cache still needed until PoS). I don't think being "smart" is the good thing to do here. I'd say the correct fix is for the miner to not generate new tasks once ttd is reached. That's a lot "dumber" and more predictable.

Reply:
Right now it only shuts down the remote sealer. I agree that it is not a great way to do this though